controller/control.go

```diff
@@ -583,6 +583,12 @@ func (c *Controller) Start(addresses ...string) error {
 		if err := c.addReplicaNoLock(newBackend, address, false); err != nil {
 			return err
 		}
+	getCloneStatus:
+		if status, _ := c.backend.GetCloneStatus(address); status != "completed" && status != "NA" {
```

Comment:
status must have 2 values at the same time?

Reply:
Here we are checking that current status of clone, it can be either `completed` or `NA`, if either one of them is not there that means sync is going on, then check it again in 2 sec interval

Reply:
thats not what the code says... or means && something else in go?

Reply:
If the replica type is clone, then we wait over here for its status to be turned to "completed".
If the replica is an original replica (not a cloned version), we wait for the status to be set to "NA(Not Applicable).

Reply:
status != "completed" && status != "NA" this reads to me that the status should be NOT completed and NOT NA, so what state does this imply? this is a state that does not exist right now?

Reply:
It means the state has not been set till now by the replica. i.e the status returned is "".
Or, inprogress and error states have also been added.
A new commit has been added where inprogress/""/error states are being checked instead of above logic to be more clearer.